### PR TITLE
feat: add xero piece

### LINKED ISSUE #904: Piece: Xero Accounting
**Describe your usecase.**
- Create/Update Contact
- Create/Update Invoice

### PR DESCRIPTION
**Pull request by kanarelo**
## What does this PR do?

Fixes #904

### CODE REVIEW

File: packages/pieces/xero/src/lib/actions/create-invoice.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        "UpdatedDateUTC": "/Date(1679982403190+0000)/",
+        "CurrencyCode": "KES"
+      }
+    ]
+  },
+  props: {
+    authentication: props.authentication,
+    tenant_id: props.tenant_id,
+    invoice_id: props.invoice_id,
+    contact_id: props.contact_id(true),
```
Comment (abuaboud) [inline packages/pieces/xero/src/lib/actions/create-invoice.ts]: Let's replace contact_id with contact_name (required) and contact_email (optional).

File: packages/pieces/xero/src/lib/actions/create-invoice.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+      defaultValue: {
+        AccountCode: 200,
+        Quantity: 0,
+        UnitAmount: 0,
+        LineAmount: 0,
+        TaxType: "NONE",
+        Description: "",
+      }
+    }),
+    date: Property.ShortText({
```
Comment (abuaboud) [inline packages/pieces/xero/src/lib/actions/create-invoice.ts:135]: This should be optional

File: packages/pieces/xero/src/lib/actions/create-invoice.ts [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      displayName: "Date Prepared",
+      description: "Date the invoice was created. Format example: 2019-03-11",
+      required: true
+    }),
+    due_date: Property.ShortText({
```
Comment (abuaboud) [inline packages/pieces/xero/src/lib/actions/create-invoice.ts:140]: Please add default value, ideally we should replace it with date picker.

File: packages/pieces/xero/src/lib/common/props.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    authUrl: "https://login.xero.com/identity/connect/authorize",
+    tokenUrl: "https://identity.xero.com/connect/token",
+    required: true,
+    scope: [
+      'accounting.contacts',
+      'accounting.transactions'
+    ]
+  }),
+  tenant_id: Property.Dropdown({
+    displayName: 'Tenant ID',
```
Comment (abuaboud) [inline packages/pieces/xero/src/lib/common/props.ts]: Let's rename it to `Organization`

File: packages/pieces/xero/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+
+import { createPiece } from "@activepieces/framework";
+import packageJson from "../package.json";
+import { xeroCreateContact } from "./lib/actions/create-contact";
+import { xeroCreateInvoice } from "./lib/actions/create-invoice";
+
+export const xero = createPiece({
+  name: "xero",
+  displayName: "Xero",
+  logoUrl: "https://upload.wikimedia.org/wikipedia/en/9/9f/Xero_software_logo.svg",
```
Comment (abuaboud) [inline packages/pieces/xero/src/index.ts]: Let's replace this to `https://cdn.activepieces.com/pieces/xero.png`

**Review by abuaboud — CHANGES_REQUESTED** [review]

File: packages/pieces/xero/src/lib/actions/create-invoice.ts (hunk 2 above)
Comment (kanarelo) [reply packages/pieces/xero/src/lib/actions/create-invoice.ts:135]: Ok

**Comment by kanarelo** [comment]
@abuaboud cleared. You can test now

**Pull request merged**